src/backend-api/Controllers/JobController.cs

```diff
@@ -158,6 +180,48 @@ private MimeMessage CostBreakDownMessage(MimeMessage message)
             return message;
         }
 
+        /* weeklyCostOfItem<T>() calculates the past week's cost for an item. 
+         *   If an item was bought within the past week, then it will be added.
+         *   If an item was renewed in the last week, it will be added and CostPerYear normalized
+         *   to the length of the renewal period (i.e. if something is renewed every month, the
+         *   cost the past week will be CostPerYear/12)
+         * Returns: deciaml? which is the past week's cost of the item.
+         */
+        private decimal? weeklyCostOfItem<T>()
+            where T : class, IPurcahseRenewal, ISoftDeletable
+        {
+            DateTime today = DateTime.Today;
+            DateTime lastWeek = today.AddDays(-7);
+            decimal? costPrograms = _context.Set<T>()
+                    // Don't add any costs of deleted things.
+                    .Where(x => !x.IsDeleted)
+                    .Select(x => new
+                    {
+                        // Find the previous renewal date, and also select vars needed in future.
+                        PreviousRenewal = x.RenewalDate != null ?
+                        x.RenewalDate.Value.AddMonths(x.MonthsPerRenewal != null ? -x.MonthsPerRenewal.Value : -999) : new DateTime(1800),
```

Comment:
Perhaps explain why we are setting previous renewal to 100 years ago if renewal date is null